src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -259,40 +259,49 @@ ss::future<try_abort_reply> tx_gateway_frontend::do_try_abort(
                 try_abort_reply{.ec = tx_errc::stm_not_found});
           }
 
-          return stm->barrier().then([&self, stm, pid, tx_seq, timeout](
-                                       bool ready) {
-              if (!ready) {
-                  return ss::make_ready_future<try_abort_reply>(
-                    try_abort_reply{.ec = tx_errc::unknown_server_error});
-              }
-              auto tx_id_opt = stm->get_id_by_pid(pid);
-              if (!tx_id_opt) {
-                  return ss::make_ready_future<try_abort_reply>(
-                    try_abort_reply{.aborted = true, .ec = tx_errc::none});
-              }
-              auto tx_id = tx_id_opt.value();
-
-              auto tx_opt = stm->get_tx_by_id(tx_id);
-              if (!tx_opt) {
-                  return ss::make_ready_future<try_abort_reply>(
-                    try_abort_reply{.aborted = true, .ec = tx_errc::none});
-              }
-              auto tx = tx_opt.value();
-              if (tx.pid != pid || tx.tx_seq != tx_seq) {
-                  return ss::make_ready_future<try_abort_reply>(
-                    try_abort_reply{.aborted = true, .ec = tx_errc::none});
-              }
-
-              return stm->get_tx_lock(tx_id)
-                ->with(
-                  timeout,
-                  [&self, stm, tx_id, pid, tx_seq, timeout]() {
-                      return self.do_try_abort(
-                        stm, tx_id, pid, tx_seq, timeout);
-                  })
-                .handle_exception_type([](const ss::semaphore_timed_out&) {
-                    return try_abort_reply{.ec = tx_errc::unknown_server_error};
-                });
+          return stm->read_lock().then([&self, stm, pid, tx_seq, timeout](
+                                         ss::basic_rwlock<>::holder unit) {
```

Comment:
can we split out clang-format changes for this commit? it looks like the functional changes are wrapping a few blocks in locks, but it's hard to tell since the diff ends up being quite large.

I think this comment applies to both the 2nd and 3rd commits.

Reply:
splitted